taskboard/client.py

```diff
@@ -0,0 +1,33 @@
+from .exceptions import TaskBoardError
+from .backend_interface import TaskBoardBackend
+from typing import List, Dict, Any
+
+class TaskBoardClient:
+    def __init__(self, backend: TaskBoardBackend):
+        """
+        Initialize the TaskBoardClient with a backend.
+        :param backend: Instance implementing TaskBoardBackend interface.
+        """
+        if not backend:
+            raise TaskBoardError("backend must be provided")
+        self.backend = backend
```

Comment:
why are you doing the extra hop again?
there should be no proxy / containment pattern here.
the implementation should be inheriting the interface and implementing the methods.

Reply:
I see. Changed interface definition to define abstract base class which will be inherited by the implementation's class.